test/distributed/org/apache/cassandra/distributed/test/ReplicaFilteringProtectionTest.java

```diff
@@ -50,30 +52,43 @@
     private static final int ROWS = 3;
 
     private static Cluster cluster;
+    private static String skipMessage;
 
     @BeforeClass
     public static void setup() throws IOException
     {
-        cluster = init(Cluster.build()
-                              .withNodes(REPLICAS)
-                              .withConfig(config -> config.set("hinted_handoff_enabled", false)
-                                                          .set("commitlog_sync", "batch")
-                                                          .set("num_tokens", 1)).start());
-
-        // Make sure we start w/ the correct defaults:
-        cluster.get(1).runOnInstance(() -> assertEquals(DEFAULT_WARN_THRESHOLD, StorageService.instance.getCachedReplicaRowsWarnThreshold()));
-        cluster.get(1).runOnInstance(() -> assertEquals(DEFAULT_FAIL_THRESHOLD, StorageService.instance.getCachedReplicaRowsFailThreshold()));
+        try
+        {
+            cluster = init(Cluster.build()
+                                  .withNodes(REPLICAS)
+                                  .withConfig(config -> config.set("hinted_handoff_enabled", false)
+                                                              .set("commitlog_sync", "batch")
+                                                              .set("num_tokens", 1)).start());
```

Comment:
Why does this test need num_tokens set to 1 like this? Seems like this is the cause of needing all the song and dance around capturing the skipMessage etc. Also seems somewhat unique to this test, but I can't think of any reason replica filtering as a mechanism wouldn't need to also be tested in a vnode enabled environment? (caveat: haven't read through all the tests here in detail; spent time reviewing the logic around capturing the skip reasoning etc)

Reply:
> Why does this test need num_tokens set to 1 like this?

not author so can't justify; my guess is "I want to be sure we don't use vnode", @maedhroz mind commenting?

> Seems like this is the cause of needing all the song and dance around capturing the skipMessage etc

actually no... this is a junit issue -_-. If you rely on static clusters (tests reusing the same cluster) then throwing a skip error will not skip the tests, but will mark the class as failing; aka red build.

Since this class does not allow vnode, the assumption error is thrown in the static method, but we need to skip the tests, so each test needs to do the check... We could change the behavior to allow the cluster to startup with vnode, but have the tests skip if the cluster is vnode... we waste resources is the only side effect

Reply:
Ah, got it. I'd rather just leave it as is and probably put some of what you stated here in a comment in the class (note about cluster reuse and consequences, assumption in static method, etc)

Reply:
I think this was just a case of me explicitly simplifying the state of the test. There are only two nodes in this cluster, and it should work just fine with or without vnodes.

Reply:
Ah, ok, I think the other reason is that in Python dtest space, `ReplicaSideFiltering` does test w/ vnodes.

Reply:
>  put some of what you stated here in a comment in the class 

so the comment at the catch, just moved to class level?

Reply:
pushed bigger doc at the skipMessage field explaining why its a thing

Reply:
I reverted the skip logic... I remember it was causing issues but it no longer is and CASTest is the same pattern... the junit xml shows the class as skipped...

Reply:
I removed `num_tokens` and test passes... let the vnodes begin!